File: Frontend/lib/src/AdminRecord.dart
import 'Threat.dart';

class AdminRecord {
  String id;
  DateTime reportTime;
  DateTime analysePeriodStartTime;
  DateTime analysePeriodEndTime;
  num threatLevel;
  num numberOfThreatRecords;
  List<Threat> listOfThreats;

  AdminRecord(this.id, this.reportTime, this.analysePeriodStartTime, this.analysePeriodEndTime, this.threatLevel,
      this.numberOfThreatRecords, this.listOfThreats);



}

File: InternalApi/src/main/java/com/cloudpartners/csaware/model/AdminRecord.java
package com.cloudpartners.csaware.model;

import com.google.gson.Gson;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class AdminRecord {

	private String id;
	private LocalDateTime reportTime;
	private LocalDateTime analysePeriodStartTime;
	private LocalDateTime analysePeriodEndTime;
	private int threatLevel;
	private int numberOfThreatRecords;
	private List<ThreatRecord> listOfThreats;

	public String toJSON() {
		Gson gson = new Gson();
		return gson.toJson(this);
	}
}
